Neos.Media/Classes/Domain/Service/ThumbnailService.php

```diff
@@ -92,11 +92,6 @@ public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $conf
             if ($asset->getWidth() === null && $asset->getHeight() === null) {
                 return $asset;
             }
-            $maximumWidth = ($configuration->getMaximumWidth() > $asset->getWidth()) ? $asset->getWidth() : $configuration->getMaximumWidth();
-            $maximumHeight = ($configuration->getMaximumHeight() > $asset->getHeight()) ? $asset->getHeight() : $configuration->getMaximumHeight();
-            if ($configuration->isUpScalingAllowed() === false && $maximumWidth === $asset->getWidth() && $maximumHeight === $asset->getHeight()) {
-                return $asset;
-            }
```

Comment:
The `ImageResizeAdjustment` also checks in a few places if the size can/should be changed or not. 
The result of my change would of course create separate db entries for new thumbnails but wouldn't create new images if the sha1 doesn't change.

I'm not sure if there is a use case where this could be a actual problem. If so, then this should be adjusted further.

Having a global option `forceThumbnailGeneration` isn't quite right, as it doesn't actually force the creation of a separate image without further code changes. 

I'm working on a actual new feature which allows to extend the `adjustments` array with custom post-processors. This then allows image optimization, filtering and whatever the integrator wants on a per image basis. But for that all images have to go into the refresh function and every adjustment has to decide by itself if it's applied or not.

Of course a `forceImageProcessing` global config could help there too, but it didn't feel right for me.
So more opinions would be great.